src/CircuitBreakerPolicy.ts

```diff
@@ -167,7 +198,7 @@ export class CircuitBreakerPolicy implements IPolicy {
         return this.execute(fn);
 
       case CircuitState.Open:
-        if (Date.now() - state.openedAt < this.options.halfOpenAfter) {
+        if (Date.now() - state.openedAt < (this.halfOpenAfterBackof?.duration ?? 0)) {
```

Comment:
The `halfOpenAfterBackoff` should be stored in the `state` to avoid the null assertion (which should never happen)

Reply:
Good point! I didn't think of that.